File: app/src/main/java/com/cn29/aac/ui/feedentrydetail/vm/FeedEntryDetailViewModelFactory.java
package com.cn29.aac.ui.feedentrydetail.vm;

import com.cn29.aac.repo.feedentry.FeedEntryRepository;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;


/**
 * Created by Charles Ng on 20/9/2017.
 */

public class FeedEntryDetailViewModelFactory extends ViewModelProvider.NewInstanceFactory {
  private FeedEntryRepository feedEntryRepository;
  private int uid = 0;
  public FeedEntryDetailViewModelFactory(FeedEntryRepository feedEntryRepository,int uid) {
      this.feedEntryRepository = feedEntryRepository;
      this.uid = uid;
  }

  @Override
  public <T extends ViewModel> T create(Class<T> modelClass) {
    return (T) new FeedEntryDetailViewModel(feedEntryRepository,uid);
  }
}

File: app/src/main/java/com/cn29/aac/ui/feedentrydetail/vm/FeedEntryDetailViewModelFactory.kt
package com.cn29.aac.ui.feedentrydetail.vm

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider.NewInstanceFactory
import com.cn29.aac.repo.feedentry.FeedEntryRepository

/**
 * Created by Charles Ng on 20/9/2017.
 */
class FeedEntryDetailViewModelFactory(private val feedEntryRepository: FeedEntryRepository,
                                      uid: Int) :
        NewInstanceFactory() {
    private var uid = 0
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return FeedEntryDetailViewModel(feedEntryRepository, uid) as T
    }

    init {
        this.uid = uid
    }
}